src/modules/core/components/ItemsList/ItemsList.tsx

```diff
@@ -1,317 +0,0 @@
-import React, {
-  Fragment,
-  ReactNode,
-  useCallback,
-  useMemo,
-  useState,
-  useEffect,
-} from 'react';
-import {
-  defineMessages,
-  FormattedMessage,
-  MessageDescriptor,
-} from 'react-intl';
-import { useField } from 'formik';
-
-import Button from '~core/Button';
-import Popover, { Tooltip } from '~core/Popover';
-import { sortObjectsBy, recursiveNestChildren } from '~utils/arrays';
-
-import styles from './ItemsList.css';
-import { usePrevious } from '~utils/hooks';
-
-const MSG = defineMessages({
-  fallbackListButton: {
-    id: 'ItemsList.fallbackListButton',
-    defaultMessage: 'Open List',
-  },
-});
-
-export interface ConsumableItem {
-  disabled?: boolean;
-  disabledText?: MessageDescriptor | string;
-  id: number;
-  name: string;
-  parent?: number;
-  children?: ConsumableItem[];
-}
-
-interface Props {
-  /** Children to render and to use as a trigger for the Popover */
-  children?: ReactNode;
-
-  /** Whether the selector should be disabled */
-  disabled?: boolean;
-
-  /** Prefix to display before the individual item when rendering it */
-  itemDisplayPrefix?: string;
-
-  /** Suffix to display after the individual item when rendering it */
-  itemDisplaySuffix?: string;
-
-  /** The intial list of items to display (before collapsing it) */
-  list: ConsumableItem[];
-
-  /** Field name */
-  name: string;
-
-  /** Whether the value can be unset */
-  nullable?: boolean;
-
-  /** Callback function, called after the field value has been set */
-  onChange?: (value: number | null | undefined) => void;
-
-  /** Wheather or not to show the Popover's arrow */
-  showArrow?: boolean;
-}
-
-const displayName = 'ItemsList';
-
-const ItemsList = ({
```

Comment:
While this was used strictly on the task, it's built for being generalized.

I has a very nice piece of logic that allows you to have nested items in the list.

Please put this back in, as we still want to keep it.